pkg/operator/ceph/cluster/osd/create.go

```diff
@@ -402,7 +402,7 @@ func createDaemonOnPVC(c *Cluster, osd OSDInfo, pvcName string, config *provisio
 	return nil
 }
 
-func createDaemonOnNode(c *Cluster, osd OSDInfo, nodeName string, config *provisionConfig) error {
+func createDaemonOnNode(c *Cluster, osd *OSDInfo, nodeName string, config *provisionConfig) error {
 	d, err := deploymentOnNode(c, osd, nodeName, config)
```

Comment:
d, err := deploymentOnNode(c, &osd, nodeName, config)

Reply:
The `osd` is already a pointer type

Reply:
```
func deploymentOnNode(c *Cluster, osd OSDInfo, nodeName string, config *provisionConfig) (*appsv1.Deployment, error) {

```

but this dont look for a pointer value, instead it look for actuall OSDInfo value and not the address.


So we maybe need to decrypt the pointer to the value, which can be done by `*`
```
var t *time.Time = &time.Now()
var t2 time.Time = *t
```

Which makes it to be like 

```
d, err := deploymentOnNode(c, *osd, nodeName, config)
```